### PR TITLE
Add support for latest LSP4J 0.8.1

### PR DESCRIPTION
**Pull request by gayanper**
Upgrade the library code to support latest lsp4j so that support for
latest LSP spec can be added. Also fixed some possible NPEs

### CODE REVIEW

**Comment by NipunaRanasinghe** [comment]
@nixel2007 @matthewmichihara @dinvlad @sonicning Since this lsp4j version has breaking API changes of LSP related to **goto definition** and **signature help**, could you please make sure your language servers are LSP 3.14.0 (lsp4j 0.8.0) compatible and please reach out if not :)

**Comment by nixel2007** [comment]
@NipunaRanasinghe BSL LS doesn't have GoTo definition and signature help capabilities yet.

Currently we are targeted to lsp4j 0.7.2 (manualy, in gradle file) but it's ok to upgrade it.

**Comment by NipunaRanasinghe** [comment]
@nixel2007 Sure thanks! :)

**Comment by gayanper** [comment]
@NipunaRanasinghe sure lets see what others think ? Or may be we can branch out from a release before we do the upgrade WDYT ?

**Comment by sonicning** [comment]
@gayanper @NipunaRanasinghe Sorry for the late reply, I have tested it with Clangd v8.0, it is ok on goto definition and signature help. LGTM

**Comment by nixel2007** [comment]
We also have upgraded to lsp4j 0.8.**1**

File: build.gradle [hunk 1]
```diff
@@ -21,7 +21,7 @@ intellij {
 }
 
 dependencies {
-    compile group: 'org.eclipse.lsp4j', name: 'org.eclipse.lsp4j', version: '0.6.0'
+    compile group: 'org.eclipse.lsp4j', name: 'org.eclipse.lsp4j', version: '0.8.0'
```
Comment (nixel2007) [inline build.gradle]:
```suggestion
    compile group: 'org.eclipse.lsp4j', name: 'org.eclipse.lsp4j', version: '0.8.1'
```

Comment (gayanper) [reply build.gradle]: I'll add that if all are ok

Comment (NipunaRanasinghe) [reply build.gradle]: +1 for 0.8.1

**Comment by gayanper** [comment]
@NipunaRanasinghe  Ready to merge

**Pull request merged**